### ISSUE TITLE
id wrong set in relation

### ISSUE THREAD

**Issue by coaradsupp** [staudenmeir/eloquent-has-many-deep#63]
hi, tnx for this package
i use this format for better explaine:

```

//Office Model

/**
	 * relation of office with reserve.
	 * @return HasManyThrough
	 */
	public function reserves() {
		return $this->hasManyThrough( Reserve::class , Worktime::class );
	}

	/**
	 * relation of office with worktime.
	 * @return HasMany
	 */
	public function worktimes() {
		return $this->hasMany( Worktime::class );
	}
/////////////////////////////////////////////////////////////////////////////

//Worktime Model

/**
	 * relation of worktime with office.
	 * @return BelongsTo
	 */
	public function office() {
		return $this->belongsTo( Office::class );
	}

	/**
	 * relation of worktime with reserve.
	 * @return HasMany
	 */
	public function reserves() {
		return $this->hasMany( Reserve::class );
	}

/////////////////////////////////////////////////////////////////////////////

//Reserve Model

	/**
	 * relation of reserve with office.
	 * @return HasOneDeep
	 */
	public function office() {
		return $this->hasOneDeepFromRelations( $this->worktime(), (new Worktime)->office());
	}

	/**
	 * relation of worktime with office.
	 * @return BelongsTo
	 */
	public function worktime() {
		return $this->belongsTo( Worktime::class );
	}
/////////////////////////////////////////////////////////////////////////////


```

in this relation

$reserve = Reserve::find(43);
dd($reserve->load('office'));

id of office wrong
worktime_id  set to office id

how solve this problem?????
![Capture](https://user-images.githubusercontent.com/60090921/75242874-28f93b80-57de-11ea-8146-5e54ef5b8b2f.PNG)

**Comment by staudenmeir** [comment]
What's the result of `dd(Reserve::find(43)->getAttributes(), Reserve::find(43)->worktime->getAttributes());`?

**Comment by coaradsupp** [comment]
thank you for your response


this result
![Capture1](https://user-images.githubusercontent.com/60090921/75312562-87bac580-586f-11ea-97d0-ef7716fb0661.PNG)

**Comment by staudenmeir** [comment]
I'll look into it.

**Comment by staudenmeir** [comment]
What version of the package are you using (`composer show`)?

What's the result of this snippet?

```php
\DB::enableQueryLog();
Reserve::find(43)->load('office');
dd(\DB::getQueryLog());
```

**Comment by coaradsupp** [comment]
this version :
"staudenmeir/eloquent-has-many-deep": "^1.7",

10 instead 43
```
\DB::enableQueryLog();
		Reserve::find(10)->load('office');
		dd(\DB::getQueryLog());
```

return this result:
![Capture](https://user-images.githubusercontent.com/60090921/75475207-a1622700-59ad-11ea-8809-069167037aa4.PNG)

**Comment by staudenmeir** [comment]
What does `composer show` show for `staudenmeir/eloquent-has-many-deep`?

What's your Laravel version?

**Comment by coaradsupp** [comment]
```
staudenmeir/eloquent-has-many-deep         v1.7     Laravel Eloquent HasManyThrough relationship with unlimited levels



laravel/framework                          v5.6.39  The Laravel Framework.
```

**Comment by staudenmeir** [comment]
This is/was a Laravel issue and has been fixed in Laravel 5.8.

If you can't upgrade, you can use one of my other packages:
https://github.com/staudenmeir/belongs-to-through

```php
class Reserve extends Model
{
    use \Znck\Eloquent\Traits\BelongsToThrough;

    public function office()
    {
        return $this->belongsToThrough(Office::class, Worktime::class);
    }
}
```

**Comment by coaradsupp** [comment]
Thans for your good and complete answers

so use this package
